grafana-stack/deploy

```diff
@@ -0,0 +1,230 @@
+print_line "Server host:                ${SERVER_HOST}"
+
+exec 2>deploy.log
+exec 1>&2
+
+set -x
+
+create_container() {
+  container=$1
+  dockerfile=$2
+  docker build -t "${container}" -f "${dockerfile}" "$(dirname ${dockerfile})"
+}
+
+push_container() {
+  container=$1
+  case "$container" in
+    gcr.io/*)
+      gcloud docker push "$container"
+      ;;
+    *)
+      docker push "$container"
+  esac
+}
+
+apply_config() {
+  config=$1
+  kubectl --kubeconfig="${KUBECONFIG}" apply -f "${config}"
+}
+
+create_pd() {
+  # This is far from perfect. The size must match what is in configuration files
+  name=$1
+  vclaim=$2
+  volume=$3
+  size=$4
+
+  ! test -z "$(gcloud compute disks list --uri ${name})" || \
+          gcloud compute disks create "${name}" --size "${size}" &&
+  apply_config "${vclaim}" &&
+  apply_config "${volume}"
+}
+
+prepare_nginx() {
+  nginx_tag=$(date +%s)
+  nginx_repo=${REGISTRY}/nginx
+}
+
+prepare_grafana() {
+  grafana_tag=$(date +%s)
+  grafana_repo=${REGISTRY}/grafana
+}
+
+prepare_influxdb() {
+  influxdb_tag=$(date +%s)
+  influxdb_repo=${REGISTRY}/influxdb
+}
+
+get_public_ip() {
+  service=$1
+
+  kubectl --kubeconfig="${KUBECONFIG}" get svc "${service}" -o=jsonpath="{.status.loadBalancer.ingress[0].ip}"
+}
+
+wait_nginx_service() {
+  for i in $(seq 6); do
+    ip=$(get_public_ip nginx-service) &&
+    if test ! -z ${ip}; then
+      print_line "Nginx service is up:" ${ip}
+      return
+    fi
+    sleep 5
+  done
+  false
+}
+
+create_grafana_configs() {
+  grafana_admin_password=$1
+
+  sed -e "s/admin-password-placeholder/${grafana_admin_password}/" grafana.template.ini >grafana.ini
+}
+
+prepare_nginx_deployment() {
+  nginx_image=$1
+  sed -e "s#nginx-image-placeholder#${nginx_image}#" nginx.template.yaml >nginx.yaml
+}
+
+prepare_grafana_deployment() {
+  grafana_image=$1
+  sed -e "s#grafana-image-placeholder#${grafana_image}#" grafana.template.yaml >grafana.yaml
+}
+
+prepare_influxdb_deployment() {
+  influxdb_image=$1
+  root_password=$2
+
+  sed \
+    -e "s#influxdb-image-placeholder#${influxdb_image}#" \
+    -e "s/influxdb-root-password-placeholder/${root_password}/" \
+    influxdb.template.yaml >influxdb.yaml
+}
+
+create_grafana_datasource() {
+  server_hostname=${1:-$(get_public_ip nginx-service)} &&
+  grafana_admin_password=$2 &&
+
+  curl -s --fail "http://admin:${grafana_admin_password}@${server_hostname}/api/datasources/name/github" ||
+  cat datasource.json | env - server_hostname=${server_hostname} envsubst | curl -X POST --fail -H 'Content-Type: application/json' -H 'Accept: application/json' "http://admin:${grafana_admin_password}@${server_hostname}/api/datasources" --data-binary @-
+}
+
+create_influxdb_user() {
+  server_hostname=${1:-$(get_public_ip nginx-service)} &&
+  influxdb_root_password=$2 &&
+
+  cat create_user | env - server_hostname=${server_hostname} influx_db_root_password=${influxdb_root_password} envsubst | sh
```

Comment:
I'm trying to keep configurations seperated from the deploy script as much as I can. 

*... looking at the file ...*

But in the end it looks more like a shell script than a configuration file. *Updating...*